src/test/suite/asciidocUtilsTest.test.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as vscode from 'vscode';
+import * as extensionFunctions from '../../extensionFunctions';
+import * as path from 'path';
+import { expect } from 'chai';
+import * as Utils from './Utils';
+
+suite('AsciiDoc Utils Test Suite', () => {
+
+	setup(() => {
+		Utils.ensureExtensionActivated();
+	});
+
+	test('open an asciidoc file', async () => {
+		const testFile = path.resolve(__dirname, '..', '..', '..', './demos/asciidoc/didact-demo.didact.adoc');
+		const content = await extensionFunctions.getDataFromFile(vscode.Uri.parse(testFile));
+		expect(content).to.not.equal(null);
+		expect(content).to.include('Ideas or want to contribute?');
+	});
+
+	test('open an asciidoc file with an include', async () => {
+		const testFile = path.resolve(__dirname, '..', '..', '..', './src/test/data/includetext.didact.adoc');
+		const content = await extensionFunctions.getDataFromFile(vscode.Uri.parse(testFile));
+		expect(content).to.not.equal(null);
+		expect(content).to.include('The quick brown fox jumps over the lazy dog.');
+	});
+
```

Comment:
I think a test with a referenced Didact file should be added to cover the use case mentioned here https://issues.redhat.com/browse/FUSETOOLS2-804?focusedCommentId=15339182&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15339182
or it is not implemented in this PR? If not, I think it can be handled in another PR.

Reply:
The 'open an asciidoc file with an include' test does test if the include is happening correctly. It does not test a deeper (i.e. subdirectory-level) include, but I am not sure that's necessary if that's what you are suggesting.

Reply:
No, this is not what i'm suggesting but it might a good test too.

What I'm suggesting is a test checking that an included Didact AsciiDoc file is well interpreted. For instance, that a Didact link inside the included file is working.

Reply:
The test that's in place looks for text that comes into the merged didact file from the included file. We really don't have a facility for going deeper than that at this point.

Reply:
Though I'm not sure how to do that level of deeper functional testing (links/etc.), I have opened a new JIRA for it. https://issues.redhat.com/browse/FUSETOOLS2-810

Reply:
if I understand well, the use case is working but writing a test is too complicated?

Reply:
manually tested. The use case is working.

Reply:
the use case of including text is being tested is tested. 

This test pulls in a didact.adoc (`includetext.didact.adoc`) with the line `include::include.asciidoc.txt[]` which points to a file that includes the text `The quick brown fox jumps over the lazy dog.`. The function getDataFromFile uses the asciidoc rendered to pull in the include file automatically into the generated HTML file that is shown in the webview. If the text is not in the rendered HTML, asciidoc is not pulling in the include file correctly.

```
test('open an asciidoc file with an include', async () => {
		const testFile = path.resolve(__dirname, '..', '..', '..', './src/test/data/includetext.didact.adoc');
		const content = await extensionFunctions.getDataFromFile(vscode.Uri.parse(testFile));
		expect(content).to.not.equal(null);
		expect(content).to.include('The quick brown fox jumps over the lazy dog.');
	});
```

I think this is a reasonable test that looks at the use case.

Reply:
> the use case of including text is being tested.

yes but not the use case of including a Didact file which is the one I requested and that Paul is expecting.
https://issues.redhat.com/browse/FUSETOOLS2-810 will provide the test for this use case.